modules/videopress/class.jetpack-videopress.php

```diff
@@ -334,6 +336,43 @@ public function add_videopress_extenstion( $extensions ) {
 
 		return $extensions;
 	}
+
+	/**
+	 * Register the Jetpack Gutenberg extension that adds VideoPress support to the Core video block.
+	 */
+	public function register_gutenberg_extension() {
+		register_block_type( 'core/video', array(
+			'render_callback' => array( $this, 'render_block_core_video_with_videopress' ),
+		) );
+	}
+
+	/**
+	 * Render the Core video block replacing the src attribute with the VideoPress URL
+	 *
+	 * @param array  $attributes Array containing the video block attributes.
+	 * @param string $content    String containing the video block content.
+	 *
+	 * @return string
+	 */
+	public function render_block_core_video_with_videopress( $attributes, $content ) {
+		if ( ! isset( $attributes['id'] ) ) {
+			return $content;
+		}
+
+		preg_match('/src="([^"]+)/', $content, $matches);
+		$original_src = $matches[ 1 ];
+		$src = $this->update_attachment_url_for_videopress( $original_src, $attributes['id'] );
+
+		return preg_replace(
+			'/src="([^"]+)/',
+			sprintf(
+				'src="%1$s',
+				esc_attr( $src )
+			),
+			$content,
+			1
+		);
```

Comment:
we're opening ourselves to an unexpected error here since the regular expression match could fail, in which case `$matches[1]` would be `null` and we don't want to replace anything at that point.

we can guard against that by either checking for the success or by using `preg_replace_callback` which is going to be inherently safe. I tried tossing together a method using `preg_replace_callback` but it got nasty trying to pass around `id` since we can't use anonymous functions or closures in Jetpack code due to the PHP version.

```php
/**
 * @see render_block_core_video_with_videopress for RegExp patter
 */
public function render_block_core_video_with_videopress( $attributes, $content ) {
	if ( ! isset( $attributes[ 'id' ] ) ) {
		return $content;
	}

	$src_pattern = '/(src=")(.+?)(")/';
	if ( 1 !== preg_match( $src_pattern, $content, $matches ) ) {
		// no pattern match, can't replace URL
		return $content;
	}

	list( $full_match, $prefix, $src, $suffix ) = $matches;

	$videopress_url = $this->update_attachment_url_for_videopress( $src, $attributes['id'] );
	if ( ! is_string( $videopress_url ) ) {
		// couldn't find a videopress URL
		return $content;
	}

	return preg_replace(
		$src_pattern,
		$prefix . esc_attr( $videoporess_src ) . $suffix,
		$content,
		/* replace-at-most */ 1
	);
}
```

Reply:
Since we don't need the original URL for retrieving the VideoPress URL I simplified the code in 0f8e96e,  so `preg_match` is no longer needed.